lib/shopify-cli/tasks/update_dashboard_urls.rb

```diff
@@ -1,27 +1,36 @@
 module ShopifyCli
   module Tasks
-    class UpdateWhitelistURL < ShopifyCli::Task
+    class UpdateDashboardURLS < ShopifyCli::Task
       def call(ctx, url:)
         @ctx = ctx
         project = ShopifyCli::Project.current
         app_type = project.app_type
-        api_key = Project.current.env.api_key
+        api_key = project.env.api_key
+        @regex = /https:\/\/([a-z0-9\-]+\.ngrok\.io)(.*)/
         result = Helpers::PartnersAPI.query(ctx, 'get_app_urls', apiKey: api_key)
         callback = app_type.callback_url
         app = result['data']['app']
-        whitelist_urls = check_urls(app['redirectUrlWhitelist'], url)
+        whitelist_urls = check_redirect_urls(app['redirectUrlWhitelist'], url)
         with_callback = check_callback_url(whitelist_urls, url, callback)
-        return if with_callback == app['redirectUrlWhitelist']
+        opt = check_application_url(app['applicationUrl'], url)
+        return if with_callback == app['redirectUrlWhitelist'] && url == app['applicationUrl']
```

Comment:
We can remove this clause `with_callback == app['redirectUrlWhitelist']` as I can't find a scenario where it isn't false. `check_redirect_urls` already constructs a list of updated URLs with the new, meaning that the `grep.empty?` in `check_callback_urls` will never be true.

The only case we need that check is if they have a custom domain and the user answers no to the prompt, but I'd rather add that logic to `check_redirect_urls` and have a single method that constructs the whitelist urls for all cases. Mine looks like this:

```
      def construct_redirect_urls(urls, new_url, callback)
        if urls.grep(/#{new_url}#{callback}/).empty?
          urls.push("#{new_url}#{callback}")
        end
        urls.map do |url|
          if (match = url.match(@regex))
            "#{new_url}#{match[2]}"
          else
            url
          end
        end
      end
```